tt_eager/tt_dnn/queue_api.hpp

```diff
@@ -0,0 +1,79 @@
+void EnqueueAllocateDeviceBuffer(Queue&, DeviceBuffer&);
+void EnqueueAllocateHostBuffer(Queue&, HostBuffer&);
+
+
+void EnqueueDeallocate(Queue&, Tensor&);
+void EnqueueReallocate(Queue&, Tensor&);
```

Comment:
What does _EnqueueDeallocate_ and _EnqueueReallocate_ do?
Cos I can't find any usage example in the below.

Reply:
- `EnqueueDeallocate` would enqueue an operation for deallocating the tensor's storage.
- `EnqueueDeallocate` would enqueue an operation for reallocating the tensor's storage.
And we are still thinking about this API because we could potentially use `EnqueueOperation` for both of these

Reply:
What does 'reallocate' do?

Reply:
I'm not sure either, but do you need these `EnqueueDeallocate`/`EnqueueReallocate` APIs? I was under the impression that you pre-allocated a pool of memory beforehand and used/recycled these Tensors/buffers during runtime

Reply:
Per @razorback3 , this is not first have after @arakhmati clarified what the reallocate do. Will be planned after enabling T5/GPT2.